### DISCUSSION TITLE
How to take receipt of an 8-bit string?

### DISCUSSION THREAD

**Discussion by drauschenbach** [mlua-rs/mlua#346] [Q&A] [answered]
I'm trying to take receipt of a LuaSocket style of binary string (buffer), but can't find a way to avoid utf8 parsing. It looks like `FromLua` wants my `Vec<u8>` receiver to only be able to receive a value from a Lua table. I can't use a table to hold my 8-bit data, because I'm writing an API that must be 100% LuaSocket compatible.

Since all Lua strings are supposed to support 8-bit, maybe `conversion.rs` needs a tweak or two?

**Comment by khvzak** [comment, marked as answer]
If you want to receive bytes (binary string) you can use LuaString directly or try bstr::BString type. 
LuaString has ‘as_bytes’ method to get a slice of raw bytes

**Reply by drauschenbach** [reply]
Works! I can take receipt into a `bstr::String`.
